napari/_qt/qt_viewer.py

```diff
@@ -582,21 +593,22 @@ def _on_slice_ready(self, event):
         Provides updates after slicing using the slice response data.
         This only gets triggered on the async slicing path.
         """
-        responses = event.value
+        responses: Dict[weakref.ReferenceType[Layer], Any] = event.value
```

Comment:
Just a note for a separate PR: if all our `Response`s are not sister classes, maybe we can at least make a `Protocol` for them for typing purposes?

Reply:
I'm open to the idea, but I'd need to see usage where we actually use the responses in the same way. Right now, each response contains different state because each vispy layer needs to consume different state. So defining a base class or protocol just feels like typing for typing's sake, rather than any having useful semantics.